### DISCUSSION TITLE
Other language implementations

### DISCUSSION THREAD

**Discussion by alielbashir** [Trendyol/smurfs#4] [Q&A] [answered]
I would like to contribute to a Python implementation, but I'm not sure what it should look like, because unlike Go, Python has no stable native tools to make binaries for it. My questions is what's the plan for implementing hosts and clients in other languages? 

### Possible client implementation changes
1. Download source code release instead of binaries (or just clone the repo itself since no need for binary)
2. Run the client
a. Check that the host has a runtime (Python interpreter/JVM/.NET runtime/node), check the runtime version is valid (packages.json, pyproject.toml, etc.) and then run the main file
b. Assume host has Docker and run Client in Container with appropriate runtime (with exposed port)

**Comment by peacecwz** [comment, marked as answer]
Hi @alielbashir I'm happy to see your discussion. Actually, some Python GCC complier can build python code as binary. We would like to distribute it like that. I'll write documentation on How It works Host and Client Libraries, functionalities, and flows. I'll draw some flows about libraries and then we can implement them clearly. You're right If you want to implement Python side but you don't know the flow. I'll draw and document it after that we can start to implement

Thanks
